query-connector/src/app/backend/usergroup-management.ts

```diff
@@ -161,3 +161,160 @@ export async function saveUserGroupMembership(
     throw error;
   }
 }
+
+/**
+ * Retrieves the query group membership for a specific user group.
+ * @param groupId - The unique identifier of the user group.
+ * @returns A list of queries associated with the specified group.
+ */
+export async function getQueriesWithGroupStatus(
+  groupId: string,
+): Promise<Query[]> {
+  if (!(await superAdminAccessCheck())) {
+    throw new Error("Unauthorized");
+  }
+
+  const groupCheckQuery = {
+    text: `SELECT id FROM usergroup WHERE id = $1;`,
+    values: [groupId],
+  };
+  const groupCheckResult = await dbClient.query(groupCheckQuery);
+  if (groupCheckResult.rows.length === 0) {
+    return [];
+  }
+
+  const query = {
+    text: `
+      SELECT q.id, q.query_name, g.name AS group_name,
+             COALESCE(qg.id, '') AS membership_id,
+             CASE 
+               WHEN qg.query_id IS NOT NULL THEN TRUE
+               ELSE FALSE
+             END AS is_member
+      FROM query q
+      LEFT JOIN usergroup_to_query qg 
+        ON q.id = qg.query_id AND qg.usergroup_id = $1
+      LEFT JOIN usergroup g 
+        ON qg.usergroup_id = g.id
+      ORDER BY q.query_name;
+    `,
+    values: [groupId],
+  };
+
+  const result = await dbClient.query(query);
+
+  return result.rows.map((row) => ({
+    id: row.id,
+    name: row.query_name,
+    userGroupMemberships: [
+      {
+        id: row.membership_id,
+        usergroup_id: groupId,
+        group_name: row.group_name,
+        is_member: row.is_member,
+      },
+    ],
+  }));
+}
+
+/**
+ * Adds queries to a user group.
+ * @param groupId - The unique identifier of the user group.
+ * @param queryIds - The unique identifiers of the queries to add.
+ * @returns The query IDs of the queries added to the group.
+ */
+export async function addQueriesToGroup(
+  groupId: string,
+  queryIds: string[],
+): Promise<Query[]> {
+  if (queryIds.length === 0) return [];
+
+  const ids = queryIds.map((queryId) => `${queryId}_${groupId}`);
+  const insertQuery = {
+    text: `
+      INSERT INTO usergroup_to_query (id, query_id, usergroup_id)
+      SELECT * FROM unnest($1::text[], $2::uuid[], $3::uuid[])
```

Comment:
TIL of some SQL keywords that I didn't know about before! Nice 😎

Reply:
Whoa this is fancy!!